src/Config.js

```diff
@@ -113,6 +117,36 @@ export class Config {
     }
   }
 
+  static validatePasswordPolicy(passwordPolicy) {
+    if (passwordPolicy) {
+      if (passwordPolicy.resetTokenValidityDuration !== undefined && (typeof passwordPolicy.resetTokenValidityDuration !== 'number' || passwordPolicy.resetTokenValidityDuration <= 0)) {
+        throw 'passwordPolicy.resetTokenValidityDuration must be a positive number';
+      }
+
+      if(passwordPolicy.validator && typeof passwordPolicy.validator !== 'string' && !(passwordPolicy.validator instanceof RegExp) && typeof passwordPolicy.validator !== 'function' ) {
+        throw 'passwordPolicy.validator must be a RegExp, a string or a function.';
+      }
+    }
+  }
+
+  // if the passwordPolicy.validator is configured as a string or regex then convert it to a function to process the pattern
+  static setupPasswordValidator(passwordPolicy) {
+    if (passwordPolicy && passwordPolicy.validator) {
+      if (typeof passwordPolicy.validator === 'string') {
```

Comment:
I would say both should be required if both are provided. So if you a provide a callback and a pattern, then the password must pass both to be considered valid. Either that or throw if both are provided.